### PR TITLE
[Task 07] Cast Screen

### PR DESCRIPTION
**Pull request by hungdnv-1896**
## Related Tickets

- [Task 07] Cast Detail Screen

## What's this PR do ?

- Implement CastDetailScreen

## screenshot (optional)
Simulator Screen Shot - iPhone Xʀ - 2019-06-07 at 02 25 01
Simulator Screen Shot - iPhone Xʀ - 2019-06-07 at 02 25 23

## Notes
*(Other notes)*

### CODE REVIEW

File: MovieDB/Scenes/CastDetail/CastDetailViewModel.swift [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+        let activityIndicator = ActivityIndicator()
+        let errorTracker = ErrorTracker()
+        
+        let castDetail = input.loadTrigger
+            .flatMapLatest { _ -> Driver<Cast> in
+                self.useCase.getCastDetail(id: self.cast.id)
+                    .trackActivity(activityIndicator)
+                    .trackError(errorTracker)
+                    .asDriverOnErrorJustComplete()
+        }
```
Comment (dongtg-1485) [inline MovieDB/Scenes/CastDetail/CastDetailViewModel.swift]: lùi 1 tab

File: MovieDB/Domain/Model/Cast.swift [hunk 2]
```diff
@@ -13,6 +13,11 @@ struct Cast {
     var name: String
     var id: Int
     var profilePath: String
+    var gender: Gender
+    var biography: String
+    var placeOfBirth: String
+    var birthday: String
```
Comment (tuan188) [inline MovieDB/Domain/Model/Cast.swift]: Date

File: MovieDB/Platform/Repositoties/CastDetailRepository.swift [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+    
+    func getListMovieByCast(id: Int, page: Int) -> Observable<PagingInfo<Movie>> {
+        let input = API.GetMovieListByCastInput(id: id, page: page)
+        return API.shared.getListMovieByCast(input)
+            .map { output in
+                guard let movies = output.movies else {
+                    throw APIInvalidResponseError()
+                }
+                return PagingInfo<Movie>(page: page, items: movies)
+        }
```
Comment (dongsonbk58) [inline MovieDB/Platform/Repositoties/CastDetailRepository.swift]: indent

File: MovieDB/Scenes/CastDetail/CastDetailViewController.swift [hunk 4]
```diff
@@ -97,7 +93,8 @@ extension CastDetailViewController {
                 $0.castNameLabel.text = castDetailModel.name
                 $0.knowForLabel.text = String(format: "Known For: %@", castDetailModel.knowFor)
                 $0.genderLabel.text = String(format: "Gender: %@", castDetailModel.gender.title)
-                $0.birthdayLabel.text = String(format: "Birthday: %@", castDetailModel.birthday)
+                $0.birthdayLabel.text = String(format: "Birthday: %@",
+                                               castDetailModel.birthday.toString(dateFormat: "dd-MM-yyyy"))
```
Comment (thanhlong12312) [inline MovieDB/Scenes/CastDetail/CastDetailViewController.swift]: convert sang string bên CastViewModel luôn nhé

File: MovieDB/Scenes/MovieDetail/MovieDetailNavigator.swift [hunk 5]
```diff
@@ -17,7 +17,7 @@ struct MovieDetailNavigator: MovieDetailNavigatorType {
     
     func toCastDetail(cast: Cast) {
         let vc: CastDetailViewController = assembler.resolve(navigationController: navigationController,
-                                                              cast: cast)
+                                                            cast: cast)
```
Comment (thanhlong12312) [inline MovieDB/Scenes/MovieDetail/MovieDetailNavigator.swift]: ident này :v

**Pull request merged**